types/tx_msg.go

```diff
@@ -85,3 +76,13 @@ type TxDecoder func(txBytes []byte) (Tx, error)
 
 // TxEncoder marshals transaction to bytes
 type TxEncoder func(tx Tx) ([]byte, error)
+
+// MsgName returns the protobuf MessageName of a sdk.Msg.
+func MsgName(msg Msg) string {
```

Comment:
Not sure we actually need this if we always go with the type URL. Which I think we should because that's what shows up in the `Tx`